apps/console/src/features/provisioning/components/outbound-provisioning-connector-setup-form.tsx

```diff
@@ -0,0 +1,381 @@
+
+            setSelectedIdp(idp?.id);
+        }
+    }, [ idpList ]);
+
+    /**
+     * Set the IDP options for the dropdown.
+     */
+    useEffect(() => {
+        if(!idpList) {
+            return;
+        }
+
+        const idpOptions: DropdownOptionsInterface[] = [];
+        let idpOption: DropdownOptionsInterface = {
+            key: -1,
+            text: "",
+            value: ""
+        };
+
+        idpList.map((idp: IdentityProviderInterface, index: number) => {
+            idpOption = {
+                key: index,
+                text: idp.name,
+                value: idp.id
+            };
+            idpOptions.push(idpOption);
+        });
+        setIdpListOptions(idpOptions);
+    }, [ idpList ]);
+
+    /**
+     * Set the selected IDP.
+     */
+    useEffect(() => {
+        if(!selectedIdp) {
+            return;
+        }
+
+        const connectorOptions: DropdownOptionsInterface[] = [];
+        let connectorOption: DropdownOptionsInterface = {
+            key: -1,
+            text: "",
+            value: ""
+        };
+
+        getIdentityProviderDetail(selectedIdp)
+            .then((response: IdentityProviderInterface) => {
+                response.provisioning.outboundConnectors.connectors.map(
+                    (connector: OutboundProvisioningConnectorInterface, index: number) => {
+                    // Check enabled connectors
+                        if (connector.isEnabled) {
+                            connectorOption = {
+                                key: index,
+                                text: connector.name,
+                                value: connector.name
+                            };
+                            connectorOptions.push(connectorOption);
+                        }
+                    });
+                setConnectorListOptions(connectorOptions);
+            });
+    }, [ selectedIdp ]);
+
+    /**
+     * The following function handles the change of the IDP.
+     *
+     * @param values - Form values.
+     */
+    const handleIdpChange = (values: Map<string, FormValue>): void => {
+        setSelectedIdp(values.get("idp").toString());
+    };
+
+    /**
+     * Sanitizes and prepares the form values for submission.
+     *
+     * @param values - Form values.
+     * @returns Prepared values.
+     */
+    const getFormValues = (values: any): Record<string, unknown> => {
+        const idpName: string = (idpListOptions.find(
+            (idp: DropdownOptionsInterface) => idp.value === selectedIdp)).text;
+
+        return {
+            blocking: isBlockingChecked ? isBlockingChecked : !!values.get("blocking").includes("blocking"),
+            connector: connector,
+            idp: idpName,
+            jit: isJITChecked ? isJITChecked : !!values.get("jit").includes("jit"),
+            rules: isRulesChecked ? isRulesChecked : !!values.get("blocking").includes("blocking")
+        };
+    };
+
+    return (
+        <Forms
```

Comment:
Is it intentional to use the deprecated forms module?

Reply:
Due to the urgency of the feature, an existing codebase has been used here. As few modifications for props were needed, couldn't reuse the existing component itself. Had to duplicate the codebase and do relevant modifications.

Reply:
Ack